### PR TITLE
Fix S2933: issue should not be raised for types marked with [Serializ…

### LINKED ISSUE #1210: Fix S2933: issue should not be raised for types marked with [Serializable]
### Description

Rule S2933 checks for fields that can be made readonly. However, for types to be serializable, fields to be serialized must not be readonly. So this produces false positives for fields in types marked [Serializable] if the field is not marked with [NonSerialized].

Remark: rule S3925 should IMHO also warn a user for readonly fields, even if the class is just marked [Serializable] and does not implement ISerializable.

### Repro steps

Create a class marked as [Serializable] similar to the following:

```csharp
[Serializable]
class Person
{
    int _birthYear;  // compliant
    [NonSerialized]
    int _notSerializedValue;  // Noncompliant
    Person(int birthYear, int notSerializedValue)
    {
        _birthYear = birthYear;
        _notSerializedValue = notSerializedValue;
    }
}
```

### Expected behavior

Rule 2933 produces a warning for the "_notSerializedValue", but no warning for "_birthYear" as this field must not be readonly for deserialization to work.

### Actual behavior

Rule 2933 produces a warning for both fields "_birthYear" and "_notSerializedValue".

### Known workarounds

None.

### Related information

* SonarC# Version = 6.7.1
* Visual Studio Version = VS 2017

### PR DESCRIPTION
**Pull request by valhristov**
…able]

Fix #1210

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldBeReadonly.cs [hunk 1]
```diff
@@ -171,4 +171,15 @@ public void Do()
...
+    [Serializable]
+    public class SerializableClass
+    {
+        private int field; // Compliant, containing class is marked with [Serializable]
+
+        public SerializableClass()
+        {
+            field = 5;
+        }
+    }
```
Comment (duncanp-sonar) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldBeReadonly.cs]: Corner case that would be trivial to add a test for: [Serializable] is not inherited so the rule should raise for child classes.
